Update to v2 libraries (untested)

MiniBioOne opened the pull request:
Resolves #2 

### Changes:

Updates the server to the latest V2 libraries.

Draft for now because I haven't tested it yet.

- [x] I have read and agreed to the [Code of Conduct](https://github.com/PretendoNetwork/Pretendo/blob/master/.github/CODE_OF_CONDUCT.md).
- [x] I have read and complied with the [contributing guidelines](https://github.com/PretendoNetwork/Pretendo/blob/master/.github/CONTRIBUTING.md).
- [x] What I'm implementing was an [approved issue](../issues?q=is%3Aopen+is%3Aissue+label%3Aapproved).
- [x] I have tested all of my changes.

Commit:
feat!(nex): Update to v2 libraries (untested)

Haven't tested these changes yet but I will once I'm home

DaniElectra commented on nex/register_common_secure_server_protocols.go:
```diff
@@ -1,35 +1,107 @@
+func generateNEXUniqueIDHandler() uint64 {
+	var uniqueID uint64
+
+	err := binary.Read(rand.Reader, binary.BigEndian, &uniqueID)
```
It would be easier to use `rand.Uint64()` from `math/rand/v2`

MiniBioOne replied:
IMO it's the same output so it shouldn't cause too much of an issue, but I'll change it if it's necessary for consistency.

DaniElectra commented on nex/register_common_secure_server_protocols.go:
```diff
@@ -1,35 +1,107 @@
+	connection := packet.Sender().(*nex.PRUDPConnection)
+	endpoint := connection.Endpoint().(*nex.PRUDPEndPoint)
+
+	lstPlayingSession := types.NewList[*mm_types.PlayingSession]()
```
```suggestion
	lstPlayingSession := types.NewList[mm_types.PlayingSession]()
```

DaniElectra commented on tri-force-heroes:
Please delete the binary file

MiniBioOne replied:
Oops, forgot to add it to gitignore.

DaniElectra commented on init.go:
```diff
@@ -21,13 +24,12 @@ func init() {
 	var err error
 
-	err = godotenv.Load()
+	err = godotenv.Load(".env")
```
Why was this changed?

MiniBioOne replied:
Honestly I completely forgot why. Must have seen it on some other server when looking at the rewrites.

DaniElectra commented on nex/authentication.go:
```diff
@@ -3,31 +3,38 @@ package nex
-		fmt.Printf("Method ID: %#v\n", request.MethodID())
-		fmt.Println("===============")
+	globals.AuthenticationServer = nex.NewPRUDPServer()
+	//globals.AuthenticationServer.SetDefaultNEXVersion(nex.NewNEXVersion(3,9,0))
+	//globals.AuthenticationServer.SetAccessKey("c1621b84")
```
Please remove the commented code

MiniBioOne replied:
Oops, sorry. Left that there so I could remind myself of the NEX version and access key.

DaniElectra commented on nex/authentication.go:
```diff
@@ -3,31 +3,38 @@ package nex
+	globals.AuthenticationEndpoint.AccountDetailsByUsername = globals.AccountDetailsByUsername
+	globals.AuthenticationServer.BindPRUDPEndPoint(globals.AuthenticationEndpoint)
+
+	globals.AuthenticationServer.LibraryVersions.SetDefault(nex.NewLibraryVersion(3, 9, 0))
```
Since this is >3.5.0, you have to enable the structure headers on the servers too (both auth and secure)

MiniBioOne replied:
This is just `globals.[Authentication/Secure]Server.ByteStreamSettings.UseStructureHeader = true`, right? (guessing on what I'm seeing on other servers.)

MiniBioOne replied:
(That is what I applied on the latest commit)

DaniElectra commented on nex/register_common_secure_server_protocols.go:
```diff
@@ -1,35 +1,107 @@
+	common_utility "github.com/PretendoNetwork/nex-protocols-common-go/v2/utility"
+	match_making "github.com/PretendoNetwork/nex-protocols-go/v2/match-making"
+	match_making_ext "github.com/PretendoNetwork/nex-protocols-go/v2/match-making-ext"
+	mm_types "github.com/PretendoNetwork/nex-protocols-go/v2/match-making/types"
```
For consistency we should use `match_making_types` instead

DaniElectra commented on nex/register_common_secure_server_protocols.go:
```diff
@@ -1,35 +1,107 @@
+	secure "github.com/PretendoNetwork/nex-protocols-go/v2/secure-connection"
+	utility "github.com/PretendoNetwork/nex-protocols-go/v2/utility"
+	"github.com/PretendoNetwork/tri-force-heroes/database"
+	local_globals "github.com/PretendoNetwork/tri-force-heroes/globals"
```
For consistency we should make `globals` mean the local to the respective project, so labeling it as `local_globals` shouldn't be needed

DaniElectra commented on nex/register_common_secure_server_protocols.go:
```diff
@@ -1,35 +1,107 @@
-	nattraversal.NewCommonNATTraversalProtocol(globals.SecureServer)
+	secureProtocol := secure.NewProtocol()
+	local_globals.SecureEndpoint.RegisterServiceProtocol(secureProtocol)
+	secure := common_secure.NewCommonProtocol(secureProtocol)
+	secure.CreateReportDBRecord = func(pid types.PID, reportID types.UInt32, reportData types.QBuffer) error {
```
```suggestion
	commonSecureProtocol := common_secure.NewCommonProtocol(secureProtocol)
	commonSecureProtocol.CreateReportDBRecord = func(pid types.PID, reportID types.UInt32, reportData types.QBuffer) error {
```

For consistency

Commit:
fix(nex): Apply @DaniElectra's suggestions.

This commit removes the binary (and adds it to .gitignore), renames local_globals to just globals for consistency (same with mm_types -> match_making_types), adds structure headers and fixes syntax errors.

MiniBioOne commented:
I've tested this and it seems to go online? Don't have anyone to test with though.

DaniElectra reviewed (approved):
Code looks good to me. Would be nice to port it to the latest versions (including the ticket validation updates), but I won't make it a blocker

MiniBioOne commented:
> Code looks good to me. Would be nice to port it to the latest versions (including the ticket validation updates), but I won't make it a blocker

I'll probably update it later and make a new PR, but its 1AM right now so I'll do it tomorrow
